src/client/src/Client.c

```diff
@@ -94,6 +94,57 @@ STATUS checkIntermittentProducerCallback(UINT32 timerId, UINT64 currentTime, UIN
     return retStatus;
 }
 
+STATUS configureClientWithRetryStrategy(PKinesisVideoClient pKinesisVideoClient, KVS_RETRY_STRATEGY_TYPE kvsRetryStrategyType) {
+    ENTERS();
+    PRetryStrategy pRetryStrategy = NULL;
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(pKinesisVideoClient != NULL, STATUS_NULL_ARG);
+
+    switch (kvsRetryStrategyType) {
+        case KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT:
+            pKinesisVideoClient->kVSRetryStrategy.createRetryStrategyFn = exponentialBackoffRetryStrategyCreate;
+            pKinesisVideoClient->kVSRetryStrategy.freeRetryStrategyFn = exponentialBackoffRetryStrategyFree;
+            pKinesisVideoClient->kVSRetryStrategy.executeRetryStrategyFn = exponentialBackoffRetryStrategyBlockingWait;
+            break;
+        case KVS_RETRY_STRATEGY_DISABLED:
+        default:
+            pKinesisVideoClient->kVSRetryStrategy.createRetryStrategyFn = NULL;
+            pKinesisVideoClient->kVSRetryStrategy.freeRetryStrategyFn = NULL;
+            pKinesisVideoClient->kVSRetryStrategy.executeRetryStrategyFn = NULL;
+    }
+
+    if (pKinesisVideoClient->kVSRetryStrategy.createRetryStrategyFn != NULL) {
+        CHK_STATUS(pKinesisVideoClient->kVSRetryStrategy.createRetryStrategyFn(
+            NULL /* use default retry strategy configuration */, &pRetryStrategy));
+        pKinesisVideoClient->kVSRetryStrategy.pRetryStrategy = pRetryStrategy;
+    }
+
+CleanUp:
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS freeClientRetryStrategy(PKinesisVideoClient pKinesisVideoClient) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(pKinesisVideoClient != NULL, STATUS_NULL_ARG);
+    CHK(pKinesisVideoClient->kVSRetryStrategy.pRetryStrategy != NULL, STATUS_SUCCESS);
+
+    if (pKinesisVideoClient->kVSRetryStrategy.freeRetryStrategyFn != NULL) {

```

Comment:
Combine both in a single line

CHK(pKinesisVideoClient->kvsRetryStrategy.pRetryStrategy != NULL && pKinesisVideoClient->kvsRetryStrategy.FreeRetryStrategyFn != NULL, retStatus);

Reply:
done